verifiable-db/src/query/merkle_path.rs

```diff
@@ -84,6 +111,71 @@ where
     /// Recomputed root for the Merkle path
     pub(crate) root: HashOutTarget,
 }
+#[derive(Clone, Debug)]
+/// Target containing data about a neighbor of a node (neighbor can be
+/// either the predecessor or the successor of a node)
+pub struct NeighborInfoTarget {
+    /// Boolean flag specifying whether the node has the given neighbor
+    is_found: BoolTarget,
+    /// Boolean flag specifying whether the neighbor is in the path from the
+    /// given node up to the root
+    is_in_path: BoolTarget,
+    /// Value of the neighbor (if the neighbor exists, otherwise a dummy value can be employed)
+    value: UInt256Target,
+    /// Hash of the neighbor node (if the neighbor exists, otherwise a dummy value can be employed)
+    hash: HashOutTarget,
+}
+
+impl NeighborInfoTarget {
+    const NUM_ELEMENTS: usize = 2 + NUM_LIMBS + NUM_HASH_OUT_ELTS;
+}
+
+impl ToTargets for NeighborInfoTarget {
+    fn to_targets(&self) -> Vec<Target> {
+        once(self.is_found.target)
+            .chain(once(self.is_in_path.target))
+            .chain(self.value.to_targets())
+            .chain(self.hash.to_targets())
+            .collect()
+    }
+}
+
+impl FromTargets for NeighborInfoTarget {
+    fn from_targets(t: &[Target]) -> Self {
+        Self {
+            is_found: BoolTarget::new_unsafe(t[0]),
+            is_in_path: BoolTarget::new_unsafe(t[1]),
+            value: UInt256Target::from_targets(&t[2..]),
+            hash: HashOutTarget::from_targets(&t[2 + NUM_LIMBS..]),
+        }
+    }
+}
+#[derive(Clone, Debug)]
+/// Set of input wires for the merkle path with neighbors gadget
+pub struct MerklePathWithNeighborsTargetInputs<const MAX_DEPTH: usize>
+where
+    [(); MAX_DEPTH - 1]:,
+{
+    pub(crate) path_inputs: MerklePathTargetInputs<MAX_DEPTH>,
+    pub(crate) end_node_inputs: EndNodeInputs,
+}
```

Comment:
Because `MerklePathWithNeighborsTargetInputs` is the data structure that contains only the input wires (i.e., the ones that need to be assigned). The information about the neighbors is an output of the gadget, that's why they appear only in the latter data structure. Might be clearer if I rename the data structure to `MerklePathWithNeighborsInputTargets` maybe?

Reply:
lol ok maybe just `MPathInputs` or stg ? It's ok to not put the full name in relation to what it does, personally i find `MerklePathWithNeighborsTargetInputs`  more confusing.

Reply:
I added the `WithNeighbors` to distinguish this from the simpler gadget `MerklePathTargetInputs` which only checks membership in the merkle path, without computing data about the predecessor/successor nodes